internal/encoder/decoder.go

```diff
@@ -82,7 +109,67 @@ func isRef(backend conf.StorageBackend, k string) bool {
 }
 
 func withPrefix(m map[string]interface{}, backend conf.StorageBackend, k string, v interface{}) string {
+
 	if backend.DecodeConfig != nil {
+		if backend.DecodeConfig.ColumnMappings != nil {
+			if mapped, ok := backend.DecodeConfig.ColumnMappings[k]; ok {
+				k = mapped
+			}
+		}
+
+		// if the value is multivalue process that first
+		isMultiValue := false
+		if backend.DecodeConfig.ListValueColumns != nil {
+			if mapped, ok := backend.DecodeConfig.ListValueColumns[k]; ok {
+				sv := v.(string)
+				if sv != "" {
+					v = strings.Split(sv, mapped)
+				}
+				isMultiValue = true
+			}
+		}
+
+		// convert the value to the correct type - if list then apply to all values
+		if backend.DecodeConfig.ColumnTypes != nil {
+			if mapped, ok := backend.DecodeConfig.ColumnTypes[k]; ok {
+				if isMultiValue {
+					sv := v.([]string)
+					if mapped == "int" {
+						iv := make([]int, 0)
+						for _, s := range sv {
+							vv, _ := strconv.Atoi(s)
+							iv = append(iv, vv)
+						}
+						v = iv
+					} else if mapped == "float" {
+						iv := make([]float64, 0)
+						for _, s := range sv {
+							vv, _ := strconv.ParseFloat(s, 64)
+							iv = append(iv, vv)
+						}
+						v = iv
+					} else if mapped == "bool" {
+						iv := make([]bool, 0)
+						for _, s := range sv {
+							vv, _ := strconv.ParseBool(s)
+							iv = append(iv, vv)
+						}
+						v = iv
+					}
```

Comment:
Maybe add an `else` to provide some log feedback if an unsupported key is provided.